java/core/src/java/org/apache/orc/impl/RecordReaderUtils.java

```diff
@@ -405,72 +410,31 @@ static void zeroCopyReadRanges(FSDataInputStream file,
     }
   }
 
-  /**
-   * Read the data from the file based on a list of ranges in a single read.
-   * @param file the file to read from
-   * @param first the first range to read
-   * @param last the last range to read
-   * @param allocateDirect should we use direct buffers
-   */
-  static void readRanges(FSDataInputStream file,
```

Comment:
Do you think we can keep this in some way, @pavibhai ? Although it's `impl` package, we know that some downstream like Hive and Flink is using our internals. If possible, we should not remove anything for safety.

Reply:
I checked out hive@master, flink@master and I don't see any references to this method.

I think it should be fine to remove this method. As you called out given that it is in impl I think we should be ok cleaning up a method that is package local.

Reply:
Thank you for checking, @pavibhai . However, it wasn't okay at all. Please see the history.

ORC-685: Add `ReaderImpl.extractFileTail` back (#570)
ORC-677: Add a deprecated legacy constructor SargApplier back (#557)
ORC-676. Add getRawDataSizeFromColIndices back to ReaderImpl (#555)
ORC-671. Add OrcTail.getStripeStatistics back for backward compatiblility

Is there any difficulty to keep this method, @pavibhai ?

Reply:
Hive and Flink were just examples here because they didn't use the last Apache ORC yet due to those breaking change from our side. Please check our Adopter page. I don't think it's safe for new `Improvement` PR to remove something in the feature release. 

https://orc.apache.org/docs/adopters.html

Even Apache ORC 2.0, I'd avoid this kind approach.

Reply:
Hive is using `<orc.version>1.6.9</orc.version>` and Flink is using `<orc.version>1.5.6</orc.version>`.
Apache ORC community had better conservative instead of moving away from our customers.

Reply:
So you are recommending that any member that is not private should be assumed to be public? Is that correct?

Reply:
Thanks for the quick call to clarify this.

I have added the deleted method back. There is some code duplication as a result of this in the method, if you are ok with rewriting this method with the new code while retaining the same interface I can do that as well.